packages/web/src/scss/components/Modal/_Modal.scss

```diff
@@ -63,23 +79,140 @@
     }
 }
 
+// To be merged into the `Modal` class.
+.Modal--composed {
+    --modal-scale: #{theme.$transition-scale-ratio};
+    --modal-translate-x: -50%;
+    --modal-translate-y: #{theme.$transition-shift-distance};
+
+    display: flex; // 1.
+    overflow: hidden; // 3.
+    visibility: hidden; // 1.
+    opacity: 0; // 1.
+
+    // 2.
+    &::backdrop {
+        background-color: transparent;
+    }
+
+    // 2.
+    &::before {
+        content: '';
+        position: fixed;
+        inset: 0;
+        z-index: -2;
+        background-color: theme.$backdrop-background-color;
+        visibility: hidden;
+        opacity: 0;
+    }
+
+    @include breakpoint.up(map.get(theme.$breakpoints, tablet)) {
+        --modal-translate-y: 50%;
+    }
+
+    @media (prefers-reduced-motion: no-preference) {
+        transition-property: visibility, opacity;
+        transition-duration: theme.$transition-duration;
+        will-change: visibility, opacity;
+
+        // 2.
+        &::before {
+            transition: inherit;
+        }
+    }
+}
+
+// `Modal__dialog` will replace the same-named class above.
+// Code duplicates are intentional to make future refactoring easier.
+.Modal--composed .Modal__dialog {
+    @include typography.generate(theme.$typography);
+
+    position: fixed;
+    bottom: 0;
+    left: 50%;
+    isolation: isolate;
```

Comment:
This is to isolate the content, right? Otherwise, as dialog is in the top-layer we should we be safe?

Reply:
OK, removed, we'll see what happens 🙂.